drivers/storage/azure/executor/azure_executor.go

```diff
@@ -0,0 +1,141 @@
+// +build !libstorage_storage_executor libstorage_storage_executor_azure
+
+package executor
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	gofig "github.com/akutz/gofig/types"
+	"github.com/akutz/goof"
+
+	"github.com/codedellemc/libstorage/api/registry"
+	"github.com/codedellemc/libstorage/api/types"
+	"github.com/codedellemc/libstorage/drivers/storage/azure"
+	azureUtils "github.com/codedellemc/libstorage/drivers/storage/azure/utils"
+)
+
+// driver is the storage executor for the azure storage driver.
+type driver struct {
+	config gofig.Config
+}
+
+func init() {
+	registry.RegisterStorageExecutor(azure.Name, newDriver)
+}
+
+func newDriver() types.StorageExecutor {
+	return &driver{}
+}
+
+func (d *driver) Init(ctx types.Context, config gofig.Config) error {
+	ctx.Info("azure_executor: Init")
+	d.config = config
+	return nil
+}
+
+func (d *driver) Name() string {
+	return azure.Name
+}
+
+// Supported returns a flag indicating whether or not the platform
+// implementing the executor is valid for the host on which the executor
+// resides.
+func (d *driver) Supported(
+	ctx types.Context,
+	opts types.Store) (bool, error) {
+	return azureUtils.IsAzureInstance(ctx)
+}
+
+// InstanceID returns the instance ID from the current instance from metadata
+func (d *driver) InstanceID(
+	ctx types.Context,
+	opts types.Store) (*types.InstanceID, error) {
+	return azureUtils.InstanceID(ctx)
+}
+
+var errNoAvaiDevice = goof.New("no available device")
+
+// NextDevice returns the next available device.
+func (d *driver) NextDevice(
+	ctx types.Context,
+	opts types.Store) (string, error) {
+	// All possible device paths on Linux instances are /dev/sd[c-p]
+	letters := []string{
+		"c", "d", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p"}
+
+	// Find which letters are used for local devices
+	localDeviceNames := make(map[string]bool)
+
+	localDevices, err := d.LocalDevices(
+		ctx, &types.LocalDevicesOpts{Opts: opts})
+	if err != nil {
+		return "", goof.WithError("error getting local devices", err)
+	}
+	localDeviceMapping := localDevices.DeviceMap
+
+	for localDevice := range localDeviceMapping {
+		re, _ := regexp.Compile(`^/dev/` +
+			azureUtils.NextDeviceInfo.Prefix +
+			`(` + azureUtils.NextDeviceInfo.Pattern + `)`)
+		res := re.FindStringSubmatch(localDevice)
+		if len(res) > 0 {
+			localDeviceNames[res[1]] = true
+		}
+	}
+
+	// Find next available letter for device path
+	for _, letter := range letters {
+		if localDeviceNames[letter] {
+			continue
+		}
+		return fmt.Sprintf(
+			"/dev/%s%s", azureUtils.NextDeviceInfo.Prefix, letter), nil
+	}
+	return "", errNoAvaiDevice
+}
+
+const procPartitions = "/proc/partitions"
+
+var devRX = regexp.MustCompile(`^sd[a-z]$`)
```

Comment:
This looks the same to me. `availLetters` is still defining [c-z], the regex is looking for [a-z], and the first comment inside of `NextDevice` says that it can be [c-p].